File: cf_sdk/src/main/java/com/example/cf_sdk/changebankapi/network/api/AuthenticationApiCreator.java
package com.example.cf_sdk.changebankapi.network.api;

import com.example.cf_sdk.changebankapi.log.Logger;

import retrofit2.Retrofit;

public class AuthenticationApiCreator implements ApiCreator<AuthenticationApi> {
    @Override
    public AuthenticationApi create(Logger logger, Retrofit retrofit) {
        return retrofit.create(AuthenticationApi.class);
    }
}

File: cf_sdk/src/main/java/com/example/cf_sdk/changebankapi/network/api/AuthenticationApiCreator.kt
package com.example.cf_sdk.changebankapi.network.api

import com.example.cf_sdk.changebankapi.log.Logger
import retrofit2.Retrofit

class AuthenticationApiCreator : ApiCreator<AuthenticationApi?> {
    override fun create(logger: Logger?, retrofit: Retrofit?): AuthenticationApi {
        return retrofit!!.create(AuthenticationApi::class.java)
    }
}